test/lock_MGN_for_REP.js

```diff
@@ -0,0 +1,189 @@
+
+/* global artifacts, web3, contract, it, before, beforeEach, after, afterEach, assert, expect, should */
+
+const MgnToken = artifacts.require('MgnToken')
+const DxLockMgnForRep = artifacts.require('DxLockMgnForRep')
+const DxReputation = artifacts.require('DxReputation')
+const BN = require('bn.js')
+
+const {
+  increaseTimeAndMine,
+  takeSnapshot,
+  revertSnapshot,
+  getTimestamp
+} = require('../src/helpers/web3helpers')(web3)
+
+contract('Locking MGN for REP', accounts => {
+  let MGN, DxLock4Rep, DxRep
+  let master
+  let snapshotId
+
+  before(async () => {
+    snapshotId = await takeSnapshot();
+    [master] = accounts
+    MGN = await MgnToken.deployed()
+    DxLock4Rep = await DxLockMgnForRep.deployed()
+    DxRep = await DxReputation.deployed()
+  })
+
+  afterEach(() => {
+    console.log('--------------------')
+  })
+
+  after(() => revertSnapshot(snapshotId))
+
+  it('can\'t lock through the scheme initially', async () => {
```

Comment:
I would consider using a GIVEN, WHEN, THEN structure approach.

Something like

```
// GIVEN: The DAO is not in the locking period yet
...

// GIVEN: And doesn't have MGN
... 

// WHEN: We try to lock MGN
...

// THEN: We get an exception
...
```

If we follow always the same structure, the test can easier to read. What do you think?

Reply:
Maybe even the second GIVEN assert is not required. It's just for illustration

Reply:
Feel free to change whatever wording you see fit.

Reply:
:) it was not wording proposal. 
Here are some links that explains what I mean. I didn't made it up, hehe

https://en.wikipedia.org/wiki/Given-When-Then

https://martinfowler.com/bliki/GivenWhenThen.html

https://www.agilealliance.org/glossary/gwt/#q=~(infinite~false~filters~(postType~(~'page~'post~'aa_book~'aa_event_session~'aa_experience_report~'aa_glossary~'aa_research_paper~'aa_video)~tags~(~'given*20when*20then))~searchTerm~'~sort~false~sortDirection~'asc~page~1)